standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreDirectSqlUtils.java

```diff
@@ -609,4 +609,10 @@ public static void throwMetaOrRuntimeException(Exception e) throws MetaException
       throw new RuntimeException(e);
     }
   }
+
+  static int getCountOfQuery(Query query, Object[] params) {
```

Comment:
nit: how about move the QueryWrapper here as well, like:
```
long start = doTrace ? System.nanoTime() : 0;
try (QueryWrapper query = new QueryWrapper(query)) {
     query.setUnique(true);
     int sqlResult = MetastoreDirectSqlUtils.extractSqlInt(query.executeWithArray(params));
     long queryTime = doTrace ? System.nanoTime() : 0;
     MetastoreDirectSqlUtils.timingTrace(doTrace, queryText, start, queryTime);
     return sqlResult;
   }
```

Reply:
Reuse `executeWithArray()` to replace this method.